Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp

```diff
@@ -36,7 +36,7 @@
 
 namespace O3DE::ProjectManager
 {
-    GemCatalogScreen::GemCatalogScreen(bool readOnly, QWidget* parent)
+    GemCatalogScreen::GemCatalogScreen(bool readOnly, QWidget* parent, DownloadController* downloadController)
```

Comment:
It looks like the downloadController is not optional because GemCatalogHeaderWidget requires one - or we need to add an `if(m_downloadController) {}` block in the `GemCatalogHeaderWidget ` constructor to avoid a crash - and provide a nullptr optional value for it so it's clear it is optional.

If we don't do that, we should move this parameter to be first and not provide a default parameter for it to make sure it is provided.

Reply:
> 

For now I've moved this to the front of the constructors and made it non-optional. I think it's probably worth doing a pass on this to move it to use AZ::Interface and something like an EBus to simplify the listeners